cli/cppcheckexecutor.cpp

```diff
@@ -158,7 +158,33 @@ bool CppCheckExecutor::parseFromArgs(CppCheck *cppcheck, int argc, const char* c
 #else
     const bool caseSensitive = true;
 #endif
-    if (!pathnames.empty()) {
+    if (!mSettings->project.fileSettings.empty()) {
+        // filter only for the selected filenames from all project files
+        std::list<ImportProject::FileSettings> newList;
+
+        for (const std::string &fname : pathnames) {
+            for (const ImportProject::FileSettings &fsetting : settings.project.fileSettings)
+            {
+                const std::string *firstName = &fname;
+                const std::string *secondName = &fsetting.filename;
+                if (firstName->size() < secondName->size())
+                {
+                    secondName = &fname;
+                    firstName = &fsetting.filename;
+                }
+
+                if (firstName->find(*secondName) != std::string::npos)
```

Comment:
hmm .. this match is not very explicit. is it your intention that if `file1.cpp` is specified then `file1.cpp` in any folder will be checked? and also for instance `test-file1.cpp`?

Reply:
What do you think about reusing the same path matching that is used for ignoring files/paths?

Reply:
I am thinking about adding a utility function in lib/path.cpp for matching paths. That can be used both for ignored paths and here. What do you think?

Reply:
I am still a bit skeptic about this matching. If you write "1.cpp" I don't think that should match "file1.cpp".

It would feel better with something more explicit. If you want to match "file1.cpp" in any subfolder than you can write "*/file1.cpp".

If you write "src" then I think it makes sense to check the flies in "src/..." but not with the path "../foo/bar/src/.."